pkg/codegen/templates/client-with-responses.tmpl

```diff
@@ -0,0 +1,227 @@
+// Client which conforms to the OpenAPI3 specification for this service.
```

Comment:
I think this is way too much code duplication.

I realize this file is hellishly complex, but I think the way to do it is to futrther parameterize the function return values within the existing template, and make a separate template file for what is unique to your response parser.

Reply:
I understand. I only duplicated this so I could maintain backwards compatibility. I'll have a go at your suggestion above, and remove this duplicate template altogether.

Thanks for the input!

Reply:
Hi @deepmap-marcinr - I've now implemented this the way you've suggested, and it seems to be working fine. Let me know if this is looking more in line with where you wanted things to go.